### DISCUSSION TITLE
fail due to ssl version Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by amg-web** [rbenv/ruby-build#1978] [Build failures]
I try to install ruby 2.7.6 on Ubuntu 22.04
build fail due to openssl version 3.0.2 does not meat the requirement:

/tmp/ruby-build.20220514164550.987.O2gQwx/ruby-2.7.6/ext/openssl/extconf.rb:111: OpenSSL >= 1.0.1, < 3.0.0 or LibreSSL >= 2.5.0 is required
        /tmp/ruby-build.20220514164550.987.O2gQwx/ruby-2.7.6/ext/openssl/extconf.rb:111:in `<top (required)>'
        ./ext/extmk.rb:214:in `load'
        ./ext/extmk.rb:214:in `block in extmake'
        /tmp/ruby-build.20220514164550.987.O2gQwx/ruby-2.7.6/lib/mkmf.rb:331:in `open'
        ./ext/extmk.rb:210:in `extmake'
        ./ext/extmk.rb:572:in `block in <main>'
        ./ext/extmk.rb:568:in `each'
        ./ext/extmk.rb:568:in `<main>'

**Comment by viceice** [comment]
Any ideas for workaround?

**Comment by mislav** [comment]
A workaround is offered in https://github.com/rbenv/ruby-build/discussions/1940#discussioncomment-2552849 and further discussion around the fix for this is https://github.com/rbenv/ruby-build/pull/1974

**Comment by amg-web** [comment]


**Reply by mislav** [reply]
You are welcome to pursue your own workaround if you do not like any of our offered solutions. However, [Ruby 2.7 cannot be linked to OpenSSL 3.0](https://bugs.ruby-lang.org/issues/18658#note-8).

**Reply by amg-web** [reply]
using precompiled binary looks better
